### PR TITLE
Fixed some more Python warnings

### PR DESCRIPTION
**Pull request by andy-maier**
**Note:** This PR is on top of PR #907.
For details, see the commit message.
Ready for review and merge.

### CODE REVIEW

File: pywbem/_recorder.py [hunk 1]
```diff
@@ -320,10 +322,18 @@ def open_file(filename, file_mode='w'):
             existing file.  if 'a' is used, the data is appended to any
             existing file.
 
+        Returns:
+
+          File-like object.
```
Comment (KSchopmeyer) [inline pywbem/_recorder.py]: What is  a file-like object

File: pywbem/cim_obj.py [hunk 2]
```diff
@@ -561,7 +565,8 @@ def _makequalifiers(qualifiers, indent):
     """
     if not qualifiers:
         return ''
-    qual_list = [q.tomof(indent + 2) for q in sorted(qualifiers.values())]
+    qual_list = [qualifiers[qn].tomof(indent + 2)
```
Comment (KSchopmeyer) [inline pywbem/cim_obj.py]: Not arguing but what is the advantage of the replacement?

File: testsuite/test_mof_compiler.py [hunk 3]
```diff
@@ -1427,35 +1451,27 @@ def test_mof_schema_roundtrip(self):
 
         repo2 = self.mofcomp2.handle
 
-        print('Start recompile file= %s' % mofout_filename)
-
+        # print('Start recompile file= %s' % mofout_filename)
```
Comment (KSchopmeyer) [inline testsuite/test_mof_compiler.py]: My only comment here is that this is a long delay for a single test.  If you are watching you begin to wonder. The start recompile gave a clue that it was really running

File: pywbem/_recorder.py (hunk 1 above)
Comment (andy-maier) [reply pywbem/_recorder.py]: "File-like object" is a Python term for an object that behaves like a file (e.g. read, write, close). Here is one article about it, but not 100% exhaustive, I guess: https://stackoverflow.com/questions/4359495/what-is-exactly-a-file-like-object-in-python

File: pywbem/cim_obj.py (hunk 2 above)
Comment (andy-maier) [reply pywbem/cim_obj.py]: The sorting was previously done on the dictionary values, i.e. on a list. That uses the ordering comparisons between the list items, i.e. CIMQualifierDeclaration objects. These objects issue a deprecation warning if ordering comparisons are used on them. The change sorts on the dictionary keys and therefore avoids the deprecation warning.

File: testsuite/test_mof_compiler.py (hunk 3 above)
Comment (andy-maier) [reply testsuite/test_mof_compiler.py]: I think it is more useful to have a clean output. py.test shows a dot for each test, and I don't think there is anything wrong is the next dot takes a minute or so to appear.

**Pull request merged**